CockpitPlugin/src/main/java/com/polidea/cockpitplugin/generator/DebugCockpitGenerator.kt

```diff
@@ -25,58 +26,138 @@ internal class DebugCockpitGenerator : BaseCockpitGenerator() {
             }
         }
         generate(file) { builder ->
             builder.addMethods(propertyMethods)
+                    .addFields(generateFields())
                     .addMethod(generateShowCockpitMethod())
         }
     }
 
+    private fun MutableList<MethodSpec>.addGetterSetterAndPropertyChangeListenerMethods(param: CockpitParam<*>) {
+        add(createGetterMethodSpecForParam(param))
+        add(createSetterMethodSpecForParam(param))
+        add(createAddPropertyChangeListenerMethodSpecForParam(param))
+        add(createRemovePropertyChangeListenerMethodSpecForParam(param))
+    }
+
+    private fun generateFields(): MutableIterable<FieldSpec> {
+        return mutableListOf<FieldSpec>().apply {
+            if (shouldGenerateColorFields) {
+                add(createColorListenerMapFieldSpec())
+                add(createColorColorMapperFieldSpec())
+                shouldGenerateColorFields = false
+            }
+        }
+    }
+
+    private fun createColorListenerMapFieldSpec(): FieldSpec {
+        return FieldSpec.builder(ParameterizedTypeName.get(mapClassName,
+                getParametrizedCockpitPropertyChangeListenerClassName(String::class.java),
+                getParametrizedCockpitPropertyChangeListenerClassName(cockpitColorClassName)),
+                "colorListenerMap", Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+                .initializer("new \$T<>()", hashMapClassName)
+                .build()
+    }
+
+    private fun createColorColorMapperFieldSpec(): FieldSpec {
+        return FieldSpec.builder(cockpitColorMapperClassName, "cockpitColorMapper",
+                Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+                .initializer("new \$T()", cockpitColorMapperClassName)
+                .build()
+    }
+
     internal fun <T : Any> createGetterMethodSpecForParam(param: CockpitParam<T>): MethodSpec {
         return createGetterMethodSpecForParamAndConfigurator(param) { builder ->
-            builder.addStatement("return (\$T) \$T.INSTANCE.getParamValue(\"${param.name}\")",
-                    mapToTypeClass(param), cockpitManagerClassName)
+            val value = param.value
+            when (value) {
+                is CockpitColor -> {
+                    val cockpitColorClass = CockpitColor::class.java
+                    builder.addStatement("\$T cockpitColor = (\$T) \$T.INSTANCE.getParamValue(\"${param.name}\")",
+                            cockpitColorClass, cockpitColorClass, cockpitManagerClassName)
+                    builder.addStatement("return cockpitColorMapper.unwrap(cockpitColor)")
+                }
+                else -> {
+                    builder.addStatement("return (\$T) \$T.INSTANCE.getParamValue(\"${param.name}\")",
+                            mapToTypeClass(param), cockpitManagerClassName)
+                }
+            }
         }
     }
 
     internal fun createSetterMethodSpecForParam(param: CockpitParam<*>): MethodSpec {
```

Comment:
Because we don't have simple types anymore and check the type in `generate` method, what about this:

`
private fun <T : Any> createSetterMethodSpecForParam(name: String, value: T): MethodSpec
`